### ISSUE TITLE
custom oauth scopes invalid

### ISSUE THREAD

**Issue by postwar-toaster** [processone/ejabberd#3556]
## Environment

- ejabberd version: 21.01
- OS: Linux (Ubuntu)
- Installed from: distro package or official deb/rpm (memory is fuzzy now)

```yaml
  "admin access":
    who:
      access:
        allow:
          - acl: loopback
          - acl: admin
          - user: "synergy@chat.jemcon.rocks"
      oauth:
          scope: "ejabberd:admin"
          access:
            allow:
              - acl: loopback
              - acl: admin
              - user: "synergy@chat.jemcon.rocks"
    what:
      - "*"
      - "!stop"
      - "!start"
 "jemcon":
    who:
      access:
        allow:
          - acl: loopback
          - acl: local
          - user: "kelvin@chat.jemcon.rocks"
      oauth:
          scope: "jemcon:user"
          access:
            allow:
              - acl: loopback
              - acl: local
              - acl: admin
              - user: "kelvin@chat.jemcon.rocks"
    what:
      - "check_account"
      - "connected_users"
      ##The rest has been abridged
```

This could just be me being stupid, but ejabberd:admin works as a scope while jemcon:user does not. Both my eyeballs and my yaml syntax highlighter aren't seeing any issues with it. Is there something else I need to do to create a custom scope? Is that not something that can be done? Or is there just a gremlin on my install?

**Comment by postwar-toaster** [comment]
I got the same error as you when I tried to get a token for the scope.

However, it seems like each command is available as a scope by itself so in theory I could just authorize for each command as an individual scope, correct? It seems like that's the intended method here. Also, ejabberd:user probably could suit my needs in this application as well. Although I have found a workaround I still have two questions.

1. Where is this documented? I only saw sasl_auth and one other default scope mentioned. Where do these other scopes come from?
2. Is it not possible to declare a custom scope or is it just not working _for me_?
